### PR TITLE
OP-1189 | Validate inventory

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
See OP-1189.

### CODE REVIEW

**Comment by mwithi** [comment]
Please resolve conflicts after #1400

File: src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java [hunk 1]
```diff
@@ -217,7 +226,74 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
...
+					if (lots != null) {
+						for (Lot lot : lots) {
+							double mainStoreQty = (double)lot.getMainStoreQuantity();
+							if (lotCodeOfMovement.equals(lotCode)) {
+								if (mainStoreQty != theoQty) { 
+									medicalInventoryRow.setTheoreticQty(mainStoreQty);
+									medicalInventoryRow = medicalInventoryRowManager.updateMedicalInventoryRow(medicalInventoryRow);
+									if (medicalInventoryRow != null) {
+										updated = true;
+										medDescriptionForUpdate = medDescriptionForUpdate + " "+medical.getDescription()+" lot code "+lot+" and qty :"+mainStoreQty+"/";
```
Comment (mwithi) [inline src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java]: Also, the last "/" is meant to be "\n" (new line)?

```suggestion
										medDescriptionForUpdate = medDescriptionForUpdate + " " + medical.getDescription() + " lot code " + lot + " and qty : " + mainStoreQty + "/";
```

File: src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java [hunk 2]
```diff
@@ -217,7 +226,74 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
...
+									}
+								}
+							} else {
+								MedicalInventoryRow medInvRow = new MedicalInventoryRow(0, mainStoreQty, mainStoreQty, inventory, medical, lot);
+								medInvRow = medicalInventoryRowManager.newMedicalInventoryRow(medInvRow);
+								if (medInvRow != null) {
+									added = true;
+									medDescriptionForNew = medDescriptionForNew + " "+medical.getDescription()+" lot code "+lot+" and qty :"+mainStoreQty+"/";
```
Comment (mwithi) [inline src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java]: Same here?

```suggestion
									medDescriptionForNew = medDescriptionForNew + " " + medical.getDescription() + " lot code " + lot + " and qty : " + mainStoreQty + "/";
```

File: src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java [hunk 3]
```diff
@@ -217,7 +226,74 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
...
+				}
+			}
+		}
+		List<OHExceptionMessage> errors = new ArrayList<>();
+		if (updated) {
+			errors.add(new OHExceptionMessage(MessageBundle.formatMessage("angal.inventory.theoreticalqtyhavebeenupdatedforsomemedical.msg", medDescriptionForUpdate)));
+		}
+		
+		if (added) {
+			errors.add(new OHExceptionMessage(MessageBundle.formatMessage("angal.inventory.theoreticalqtyhavebeenupdatedforsomemedical.msg", medDescriptionForNew)));
```
Comment (mwithi) [inline src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java]: Same bundle here?

Comment (ArnaudFonzam) [reply src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java]: yes but with different param

Comment (mwithi) [reply src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java]: ok, but they are two different things... or at least if the message is the same you could combine `medDescriptionForUpdate` and `medDescriptionForNew` in one message. Also, I was asking if you meant to list them as new lines, for which end endline character is not "/" but "\n"

File: src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java [hunk 4]
```diff
@@ -217,7 +226,74 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
...
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		List<OHExceptionMessage> errors = new ArrayList<>();
+		if (updated) {
+			errors.add(new OHExceptionMessage(MessageBundle.formatMessage("angal.inventory.theoreticalqtyhavebeenupdatedforsomemedical.msg", medDescriptionForUpdate)));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java]: if the message text includes parameters then the name of the msg should end with `.fmt.msg` per project standards.

File: src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java [hunk 5]
```diff
@@ -218,15 +233,133 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
...
+						medDescriptionForLotUpdated
+										.append("\n")
+										.append(MessageBundle.formatMessage(
+														"angal.inventory.theoreticalqtyhavebeenupdatedforsomemedical.detail.fmt.msg",
+														medicalDesc, lotInfo, theoQty, mainStoreQty, mainStoreQty - theoQty));
+					}
+				}
+			} else {
+				MedicalInventoryRow newMedicalInventoryRow = new MedicalInventoryRow(null, mainStoreQty, mainStoreQty, inventory, medical,
+								mov.getLot());
```
Comment (mwithi) [inline src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java]: Please keep it separated anyway, for more code readability:
```suggestion
				// TODO: to decide if to give control to the user about this
				double realQty = mainStoreQty;
				MedicalInventoryRow newMedicalInventoryRow = new MedicalInventoryRow(null, mainStoreQty, realQty, inventory, medical,
								mov.getLot());
```

**Review by mwithi — CHANGES_REQUESTED** [review]
LGTM. Thanks a lot! (test are failing for another problem not related to this PR)

**Review by mwithi — APPROVED** [review]

**Pull request merged**